### DISCUSSION TITLE
How to test auto connect to serial port ?

### DISCUSSION THREAD

**Discussion by anneranch** [blueman-project/blueman#2148] [Q&A]
I selected Auto-connect ->Serial Port. 
Now the options is "checked"... but the device is flashing , hence not connected.
I did closed and reopen Bluetooth manager and the device is still not connected.
I rebooted the system and the device is still not connected.
I can select Connect ->Serial port and the device gets connected. 

What am I missing in my procedure ?

**Comment by cschramm** [comment]
Auto-connect should run every minute and whenever Bluetooth becomes available, so from your description I would expect it to have initiated a connection if manually triggering one works fine.

**Comment by anneranch** [comment]
[image: image.png]
Here is what  I see on Ubuntu desktop after boot. . On the left is barely
visible Bluetooth "standard" icon. When clicked it pops-up a dialog menu
and the first entry states that  Bluetooth is on. Is that what you are
talking about,  and if so that should be enough  to "auto connect" device
which was previously selected.
It does  not "auto connect " any Bluetooth devices.

**Comment by anneranch** [comment]
ADDENDUM
I have TWO ways to execute "bluetooth manager".One is "standard" Ubuntu
desktop and the other the above mentioned Ubuntu entry in "status " bar. (
I really do not know how Ubiuntu calls that "stuff" in the upper righ
corner...)  I like to know WHAT part of Python code creates the entry to
Bluetooth manager and what part adds the "stuff" in upper right
top corner. Please provide code file / files names for the above .
Thanks

**Reply by anneranch** [reply]
ADDEDNEUM
The "Bluetooth manager"  is "executing"  blueman-manager , 
how do I find the ACTUAL  Python source code file - located in github ?

Suggested answer format 

"blueman-manager"  executes .... fill in blank ... Python source file  found here ....

**Reply by anneranch** [reply]
ADDENDUM
The "Bluetooth manager"  is "executing"  blueman-manager , 
how do I find the ACTUAL  Python source code file - located in github ?

Suggested answer format 

"blueman-manager"  executes .... fill in blank ... Python source file  found here ....

**Reply by cschramm** [reply]
The blueman-manager executable, most probably located at `/usr/bin/blueman-manager` on your system, *is* a Python file. The checked in source file that the build system uses to generate it is `apps/blueman-manager.in`.

**Comment by anneranch** [comment]
PS Can somebody tell me how to edit entry AFTER it is "send", please.

**Reply by cschramm** [reply]
https://github.com/NanoVNA-Saver/nanovna-saver/discussions/646#discussioncomment-6250712 ...

**Comment by anneranch** [comment]
OK, I got that in my local repository  ( blueman-manager.in)  . I need to
find it in the actual source code.
( I do not know how to "link "  from executable text to source code - I
assume I am missing a step to "convert " from executable to the source
code.

PS I can actually "run" some of the .py code, but never get the full GUI.


[image: image.png]
